### PR TITLE
Fixed compiled-vega resize issue

### LINKED ISSUE #365: Resizing the Vega editor panel moves it outside view
See this example

editor

### PR DESCRIPTION
**Pull request by siddhant1**
Closes #365

### CODE REVIEW

File: src/components/input-panel/compiled-spec-display/index.tsx [hunk 1]
```diff
@@ -40,6 +51,7 @@ class CompiledSpecDisplay extends React.PureComponent<Props> {
 
 function mapStateToProps(state: State, ownProps) {
   return {
+    compiledVegaPaneSize: state.compiledVegaPaneSize,
```
Comment (domoritz) [inline src/components/input-panel/compiled-spec-display/index.tsx]: Ahh cool, didn't know we had compiledVegaPaneSize in the state.

File: src/components/input-panel/compiled-spec-display/index.tsx [hunk 2]
```diff
@@ -4,20 +4,30 @@ import MonacoEditor from 'react-monaco-editor';
...
 
 type Props = ReturnType<typeof mapStateToProps>;
 
 class CompiledSpecDisplay extends React.PureComponent<Props> {
+  public editor;
+  public componentDidUpdate(prevProps) {
+    if (this.props.compiledVegaPaneSize !== prevProps.compiledVegaPaneSize) {
+      if (this.editor) {
+        this.editor.layout();
+      }
```
Comment (domoritz) [inline src/components/input-panel/compiled-spec-display/index.tsx]:
```suggestion
this.editor && this.editor.layout();
```

Comment (siddhant1) [reply src/components/input-panel/compiled-spec-display/index.tsx]: The linter gives an error on using this expression

Comment (domoritz) [reply src/components/input-panel/compiled-spec-display/index.tsx]: Oh okay

**Pull request merged**